include/boost/geometry/strategies/geographic/disjoint_segment_box_with_info.hpp

```diff
@@ -0,0 +1,259 @@
+// Boost.Geometry
+
+// Copyright (c) 2020, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_DISJOINT_SEGMENT_BOX_WITH_INFO_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_DISJOINT_SEGMENT_BOX_WITH_INFO_HPP
+
+#include <boost/geometry/core/assert.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+
+#include <boost/geometry/formulas/karney_direct_variant.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace disjoint
+{
+
+template
+<
+    typename FormulaPolicy = strategy::andoyer,
+    typename Spheroid = srs::spheroid<double>,
+    typename CalculationType = void
+>
+struct geographic_segment_box_with_info
+{
+    typedef Spheroid model_type;
+
+    inline geographic_segment_box_with_info()
+        : m_spheroid()
+    {}
+
+    explicit inline geographic_segment_box_with_info(Spheroid const& spheroid)
+        : m_spheroid(spheroid)
+    {}
+
+
+    template
+    <
+            typename Segment,
+            typename BoxPoint,
+            typename Strategy,
+            typename T
+    >
+    inline void static apply(Segment const& segment,
+                             BoxPoint const& box_point1,
+                             BoxPoint const& box_point2,
+                             Strategy const& strategy,
+                             T& result)
```

Comment:
Ok, so here the result is returned as output parameter. Probably because you implemented another version of the algorithm. However in case of distance() you probably wanted to use as much of the existing implementation as possible. Is that correct?

Reply:
I am not sure I understand the question here.

Reply:
I mean the interface of this strategy looks wierd to me.

Many `apply()` methods taking different parameters and the result returned by output argument instead of returned from the function (the return type is `void`).

Furthermore a different strategy is taken here which is something we never do in any other strategy.

Also taking min and max points of a box is something we don't do in any strategy AFAIR.

So I wanted to understand why it was done this way.

Reply:
>Furthermore a different strategy is taken here which is something we never do in any other strategy.

this is because of cycle dependencies, the disjoint_segment_box_with_info strategy cannot define a getter to a relate_segment_segment strategy so the algorithm passes that strategy. 

The rest, return types and max box points can be easily fixed to match the rest of library's design.

Reply:
This interface suggests that it's this strategy that needs the other strategy, not the algorithm. So I don't understand your point about defining a getter. If my understanding is correct and another strategy is needed by this one then it should simply be internally created and kept as a member.

Furthermore if there are some cyclic dependencies of headers because some parts of the code used here internally are defined in headers also defining some higher level code that is using this strategy then you could simply extract the common part to a different header and include it here and there.

If I'm missing something could you please describe what exactly is this cyclic dependency?